src/main/java/ro/andreiciortea/yggdrasil/template/annotation/RequestMapping.java

```diff
@@ -0,0 +1,21 @@
+package ro.andreiciortea.yggdrasil.template.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import io.vertx.core.http.HttpMethod;
+
+/**
+ * Annotation allowing REST methods on an attribute
+ *
+ * @param httpMethod any of the allowed HTTP methods (GET, POST, PUT, OPTIONS, DELETE, ...)
+ * @param path path appended to the uri of the artifact instance to call the DELETE method
+ *
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.METHOD)
+public @interface RequestMapping {
+  HttpMethod httpMethod() default HttpMethod.GET;
```

Comment:
We should avoid a direct reference to HTTP, we could reuse RequestMapping for CoAP as well. This would require a protocol-independent set of methods, but luckily HTTP and CoAP methods have an almost 1 to 1 mapping (CoAP methods are a subset of the HTTP methods and have more precise semantics).

Reply:
ok. I used the vertx HttpMethod here out of convenience. I can also specify our own method-enum and name it RequestMethod instead - would this be better?

Reply:
I've renamed it to RequestMethod of type String, this makes it more generic.